base/src/commonMain/kotlin/utils/timer/InstantTimer.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base.utils.timer
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlin.time.Duration
+import kotlin.time.ExperimentalTime
+
+@ExperimentalTime
+/** This timer finishes instantly. */
+internal class InstantTimer(override val duration: Duration) : Timer {
+    override val state: StateFlow<Timer.State> = MutableStateFlow(
```

Comment:
What it's mutable state flow?

Reply:
no, it's an immutable `StateFlow` backed by a `MutableStateFlow`. is there a more simple way to implement?

Reply:
Then you have to call `asStateFlow()` in the end

Reply:
it's being casted to a `StateFlow` by the parameter type. is there an extra benefit to use `asStateFlow()` as well?

Reply:
done

Reply:
Yes, make it strict. Using mutable with casing I still can do:
```
val mutable = state as MutableStateFlow<Timer.State>
mutable.value = Some state
```
And it will work as expected. Having asStateFlow it makes it read only forever